lib/guardian/phoenix/socket.ex

```diff
@@ -2,46 +2,49 @@ if Code.ensure_loaded?(Phoenix) do
   defmodule Guardian.Phoenix.Socket do
     @moduledoc """
     Provides functions for managing authentication with sockets.
-    Usually you'd use this on the Socket to authenticate on connection on
-    the `connect` function.
 
-    There are two main ways to use this module.
+    This module mostly provides convenience functions for storing tokens, claims and resources
+    on the socket assigns.
 
-    1. use Guardian.Phoenix.Socket
-    2. import Guardian.Phoenix.Socket
+    The main functions you'll be interested in are:
 
-    You use this function when you want to automatically sign in a socket
-    on `connect`. The case where authentication information is not provided
-    is not handled so that you can handle it yourself.
+    * `Guardian.Phoenix.Socket.authenticated?` - check if the socket has been authenticated
+    * `Guardian.Phoenix.Socket.sign_in` - Sign in a resource to a socket. Similar to `Guardian.Plug.sign_in`
+    * `Guardian.Phoenix.Socket.sign_out` - Sign out the resource on the socket
 
-    ```elixir
-    defmodule MyApp.UserSocket do
-      use Phoenix.Socket
-      use Guardian.Phoenix.Socket
+    ### Getters
 
-      # This function will be called when there was no authentication information
-      def connect(_params, socket) do
-        :error
-      end
-    end
-    ```
+    Once you're authenticated with your socket, you can use the getters
+    to fetch information about the authenticated resource for the socket.
+
+    * `Guardian.Phoenix.Socket.current_claims`
+    * `Guardian.Phoenix.Socket.current_token`
+    * `Guardian.Phoenix.Socket.current_resource`
 
-    If you want more control over the authentication of the connection, then you
-    should `import Guardian.Phoenix.Socket` and use the `sign_in` function
-    to authenticate.
+    These are the usual functions you'll want to use when dealing with authentication on sockets.
 
+    There is a bit of a difference between the usual `Guardian.Plug.sign_in` and the socket one.
+    The socket sign_in receives a token and signs in from that.
+    Please note that this is mere sugar on the underlying Guarian functions.
+
+    As an example:
     ```elixir
     defmodule MyApp.UserSocket do
       use Phoenix.Socket
-      import Guardian.Phoenix.Socket
 
-      def connect(%{"guardian_token" => token} = params, socket) do
-        case sign_in(socket, MyApp.Guardian, token) do
-          {:ok, authed_socket, guardian_params} ->
-            {:ok, authed_socket}
-          _ -> :error
+      def connect(%{"token" => token}, socket) do
+        result = Guarian.Socket.sign_in(socket, MyApp.Guardian, token)
```

Comment:
Also, maybe `authenticate` would be a more appropriate name here instead of `sign_in`?